tests/test_diagnostics.py

```diff
@@ -0,0 +1,135 @@
+            * data["income"],
+            "income_aged_72": (data["age"] == 72).astype(float)
+            * data["income"],
+        }
+    )
+
+    # Add specific characteristics to the targets to trigger warnings
+    targets = np.array(
+        [
+            (targets_matrix["income_aged_20_30"] * weights * 1000).sum(),
+            (targets_matrix["income_aged_40_50"] * weights * 1.15).sum(),
+            (targets_matrix["income_aged_71"] * weights * 1.15).sum(),
+            (targets_matrix["income_aged_71"] * weights * -1.15).sum(),
+        ]
+    )
+
+    calibrator = Calibration(
+        estimate_matrix=targets_matrix,
+        weights=weights,
+        targets=targets,
+        noise_level=0.05,
+        epochs=128,
+        learning_rate=0.01,
+        dropout_rate=0,
+    )
+
+    with caplog.at_level(logging.WARNING, logger="microcalibrate.calibration"):
+        performance_df = calibrator.calibrate()
+
+    log_text = "\n".join(record.getMessage() for record in caplog.records)
+
+    # Expected fragments
+    assert (
+        "Target income_aged_20_30" in log_text
+        and "orders of magnitude" in log_text
+    ), "Magnitude-mismatch warning not emitted."
+
+    assert (
+        "Target income_aged_71 is supported by only" in log_text
+    ), "Low-support warning not emitted."
+
+    assert (
+        "Column income_aged_72 has a zero estimate sum" in log_text
+    ), "Zero estimate sum warning not emitted."
+
+    assert (
+        "Some targets are negative" in log_text
+    ), "Negative target warning not emitted."
+
+
+def test_calibration_analytical_solution(caplog) -> None:
+    """Test the calibration process produces the expected analytical target evaluation reporting."""
+
+    # Create a mock dataset with age and income
+    random_generator = np.random.default_rng(0)
+    data = pd.DataFrame(
+        {
+            "age": random_generator.integers(18, 70, size=100),
+            "income": random_generator.normal(40000, 50000, size=100),
+        }
+    )
+    weights = np.ones(len(data))
+    targets_matrix = pd.DataFrame(
+        {
+            "income_aged_20_30": (
+                (data["age"] >= 20) & (data["age"] <= 30)
+            ).astype(float)
+            * data["income"],
+            "income_aged_40_50": (
+                (data["age"] >= 40) & (data["age"] <= 50)
+            ).astype(float)
+            * data["income"],
+        }
+    )
+    targets = np.array(
+        [
+            (targets_matrix["income_aged_20_30"] * weights).sum(),
+            (targets_matrix["income_aged_40_50"] * weights).sum(),
+        ]
+    )
+
+    calibrator = Calibration(
+        estimate_matrix=targets_matrix,
+        weights=weights,
+        targets=targets,
+        noise_level=0.05,
+        epochs=528,
+        learning_rate=0.01,
+        dropout_rate=0,
+    )
+
+    with caplog.at_level(logging.INFO, logger="microcalibrate.calibration"):
+        calibrator.assess_analytical_solution()
```

Comment:
If you return a data frame, you'll be able to check to see that the values are what you think they are. In this case they're all zero, which I think is okay for a first test, given the priority of this feature.